core/tests/wdio.conf.js

```diff
@@ -0,0 +1,414 @@
+        // an assertion fails.
+        expectationResultHandler: function(passed, assertion) {
+            // do something
+        }
+    },
+    
+    //
+    // =====
+    // Hooks
+    // =====
+    // WebdriverIO provides several hooks you can use to interfere with the test process in order to enhance
+    // it and to build services around it. You can either apply a single function or an array of
+    // methods to it. If one of them returns with a promise, WebdriverIO will wait until that promise got
+    // resolved to continue.
+    /**
+     * Gets executed once before all workers get launched.
+     * @param {Object} config wdio configuration object
+     * @param {Array.<Object>} capabilities list of capabilities details
+     */
+    //  onPrepare: function(config, capabilities) {
+      
+    // },
+    /**
+     * Gets executed before a worker process is spawned and can be used to initialise specific service
+     * for that worker as well as modify runtime environments in an async fashion.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {[type]} caps     object containing capabilities for session that will be spawn in the worker
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {[type]} args     object that will be merged with the main configuration once worker is initialized
+     * @param  {[type]} execArgv list of string arguments passed to the worker process
+     */
+    // onWorkerStart: function (cid, caps, specs, args, execArgv) {
+    // },
+    /**
+     * Gets executed just after a worker process has exited.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {Number} exitCode 0 - success, 1 - fail
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {Number} retries  number of retries used
+     */
+    // onWorkerEnd: function (cid, exitCode, specs, retries) {
+    // },
+    /**
+     * Gets executed just before initialising the webdriver session and test framework. It allows you
+     * to manipulate configurations depending on the capability or spec.
+     * @param {Object} config wdio configuration object
+     * @param {Array.<Object>} capabilities list of capabilities details
+     * @param {Array.<String>} specs List of spec file paths that are to be run
+     * @param {String} cid worker id (e.g. 0-0)
+     */
+    // beforeSession: function (config, capabilities, specs, cid) {
+    // },
+    /**
+     * Gets executed before test execution begins. At this point you can access to all global
+     * variables like `browser`. It is the perfect place to define custom commands.
+     * @param {Array.<Object>} capabilities list of capabilities details
+     * @param {Array.<String>} specs        List of spec file paths that are to be run
+     * @param {Object}         browser      instance of created browser/device session
+     */
+    before: function() {
```

Comment:
This looks similar to [this](https://github.com/oppia/oppia/blob/develop/core/tests/protractor.conf.js#L217), right? If so, the contents present inside it doesn't match. Please don't put statements irregularly instead try to check the conf file of protractor properly and then put statements accrording to that.

Reply:
No, they are not similar. In webdriverIO if the suite requested does not exists, the test will fails even before this hooks will be executed. So adding the check for suite not existing will not work. Thats why I included this check in the run_e2e_test.py line 364

Reply:
Also if some suite is mentioned in SUITES_MIGRATED_TO_WEBDRIVERIO and its not present in suites object in wdio.conf.js we recieve throws this error without adding the mentioned code:
```
Error: The suite(s) "collections", "102.0.5005.61" you specified don't exist in your config file or doesn't contain any files!
    at ConfigParser.getSpecs (/home/shivam/opensource/oppia/node_modules/@wdio/config/build/lib/ConfigParser.js:175:23)
    at /home/shivam/opensource/oppia/node_modules/@wdio/cli/build/launcher.js:51:47
    at Array.map (<anonymous>)
    at new Launcher (/home/shivam/opensource/oppia/node_modules/@wdio/cli/build/launcher.js:51:18)
    at launch (/home/shivam/opensource/oppia/node_modules/@wdio/cli/build/commands/run.js:125:22)
    at handler (/home/shivam/opensource/oppia/node_modules/@wdio/cli/build/commands/run.js:164:16)
    at Object.run (/home/shivam/opensource/oppia/node_modules/@wdio/cli/build/index.js:77:34)
```
Hence its not needed.

Reply:
If you have already added a statement in run_e2e_test.py line 364 for failing it for invalud suite name, then why there is a different error message above?